ecs-agent/acs/session/task_manifest_responder_test.go

```diff
@@ -0,0 +1,228 @@
+
+	// Test Metric publication when our responder is initialized with a mock entry factory
+	mockMetrics := testTaskManifestResponder.metricsFactory.(*mock_metrics.MockEntryFactory)
+	mockEntry := mock_metrics.NewMockEntry(ctrl)
+	mockMetrics.EXPECT().New(metrics.TaskManifestHandlingDuration).Return(mockEntry).Times(1)
+	mockEntry.EXPECT().Done(nil).Times(1)
+
+	// Create test task manifest.
+	testManifest := setupTestManifestMessage()
+
+	mockComparer := testTaskManifestResponder.taskComparer.(*mock_session.MockTaskComparer)
+	mockComparer.EXPECT().CompareRunningTasksOnInstanceWithManifest(gomock.Any()).Times(1)
+
+	// Handle the task manifest message update
+	handleManifestMessage := testTaskManifestResponder.HandlerFunc().(func(*ecsacs.TaskManifestMessage))
+	go handleManifestMessage(testManifest)
+
+	// Wait for the ack to send through our channel so we can receive it
+	manifestMessageAckSent := <-ackSent
+	require.Equal(t, aws.StringValue(manifestMessageAckSent.MessageId), aws.StringValue(testManifest.MessageId))
+
+}
+
+// TestTaskManifestSequenceNumbers will make sure the manifest can update, retrieve and catch stale
+// Manifest sequence numbers appropriately.
+func TestTaskManifestSequenceNumbers(t *testing.T) {
+
+	// Create a waitgroup to make sure the sequence number has updated
+	// before we try to access it
+	var ackWg sync.WaitGroup
+	ackWg.Add(1)
+
+	testResponseSender := func(response interface{}) error {
+		defer ackWg.Done()
+		return nil
+	}
+
+	testTaskManifestResponder := setupTestTaskManifestResponder(t, metrics.NewNopEntryFactory(), testResponseSender)
+
+	mockmmIDA := testTaskManifestResponder.messageIDAccessor.(*mock_session.MockManifestMessageIDAccessor)
+	mockmmIDA.EXPECT().SetMessageID(gomock.Any()).Times(1)
+
+	// Expect GetLatestSequenceNumber once in handleManifestMessage, and once
+	// when we validate it
+	mockSNA := testTaskManifestResponder.snAccessor.(*mock_session.MockSequenceNumberAccessor)
+	mockSNA.EXPECT().GetLatestSequenceNumber().Times(2)
+	mockSNA.EXPECT().SetLatestSequenceNumber(aws.Int64(20)).Times(1)
+
+	mockComparer := testTaskManifestResponder.taskComparer.(*mock_session.MockTaskComparer)
+	mockComparer.EXPECT().CompareRunningTasksOnInstanceWithManifest(gomock.Any()).Times(1)
+
+	// Create test task manifest.
+	testManifest := &ecsacs.TaskManifestMessage{
+		ClusterArn:           aws.String(testconst.ClusterName),
+		ContainerInstanceArn: aws.String(testconst.ContainerInstanceARN),
+		MessageId:            aws.String(testconst.MessageID),
+		Tasks: []*ecsacs.TaskIdentifier{
+			{DesiredStatus: aws.String("STOPPED"), TaskArn: aws.String(testconst.TaskARN)},
+		},
+		Timeline: aws.Int64(20),
+	}
+
+	// Handle the task manifest message update
+	handleManifestMessage := testTaskManifestResponder.HandlerFunc().(func(*ecsacs.TaskManifestMessage))
+	handleManifestMessage(testManifest)
+
+	ackWg.Wait()
+
+	// Now assert that the latest sequence number has been updated
```

Comment:
Is there a missing assertion here to assert that latest sequence number is equal to `aws.Int64(20)`? Or what is this comment for?